### PR TITLE
Blueprint Init refactor

### PR DESCRIPTION
**Pull request by Ornias1993**
# Pull Request Template

### Purpose
This PR aims to restructure the blueprint init (consolidate and simplify it).
Its main goal is to make it easier to start creating blueprints.

#### Automatic local variable creation for blueprints
A big portion of the work will be variable standardisation, both inside and outside the install script and remove the need for exclaimation mark referencing of (sub)variables in install scripts.

Currently the new variable structure is pretty mixed up. The new variable structure will be:
Config -> Install Script
`$Category_$Name_$Content -> $Content`

Example:
`jail_mariadbjail_gateway -> gateway`

Sometimes we grab data about other jails from the config file during install. This is a great way to simplify install processes. While this currently works, it would be nice streamlining this.
I suggest using:
`link_$linkname: jailname`

For example:
`link_mariadbdatabase: mariadbjail`

Grabbing the config settings from other jails (for example: the IP of a DB server) is not included in this PR.

#### standardising custom variables
When it comes to jail specific variables, such as credentials, specific settings, this will be documented and standardised.

**I suggest something like:**
For this-blueprint-only variables, just:
$value
(to be reviewed on a case-by-case basis)

**For external content:**
$origin/type_$value

For example, to specify an external mariadb database:
mariadb_database

### Please make sure you have followed the self checks below before submitting a PR:

- [X] Code is sufficiently commented
- [X] Code is indented with tabs and not spaces
- [X] The PR does not bring up any new errors
- [X] The PR has been tested
- [X] Any new files are named using lowercase (to avoid issues on case sensitive file systems)

### CODE REVIEW

**Comment by Ornias1993** [comment]
All jails are tested manually to work.

File: blueprints/mariadb/install.sh [hunk 1]
```diff
@@ -1,39 +1,23 @@
...
 
 # Make sure DB_PATH is empty -- if not, MariaDB/PostgreSQL will choke
-# shellcheck disable=SC2154
 if [ "$(ls -A "/mnt/${global_dataset_config}/${1}/db")" ]; then
 	echo "Reinstall of mariadb detected... Continuing"
 	REINSTALL="true"
 fi
 
+iocage exec "${1}" service mariadb stop
+iocage exec "${1}" rm -Rf /var/db/mysql
```
Comment (Ornias1993) [inline blueprints/mariadb/install.sh:20]: needs to be removed

File: blueprints/mariadb/install.sh [hunk 2]
```diff
@@ -1,39 +1,23 @@
...
 # Mount database dataset and set zfs preferences
 iocage exec "${1}" service mysql-server stop
 iocage exec "${1}" rm -Rf /var/db/mysql
```
Comment (Ornias1993) [inline blueprints/mariadb/install.sh:23]: We might want to do this delete thing before every mount...

Comment (Ornias1993) [reply blueprints/mariadb/install.sh:23]: Actually thats currently not feasable, it needs to be done on a case-by-case basis.

**Comment by colemamd** [comment]
I like this, vars definitely need standardization. We'll need to make sure everything is documented appropriately as well. Recommend specifying case to use as well.

**Comment by Ornias1993** [comment]
@colemamd the global vars (the ones that are there for every jail) are documented 
Already...
But yes: the others need to be pushed as well to the config of the individual jails and the syntax needs to be added to the contribution guidelines.

**Comment by Ornias1993** [comment]
@colemamd Added the requested documentation.

**Comment by Ornias1993** [comment]
I'm going to merge this, as it's a pain to rebase and it has basically been reviewed already and the review feedback (documentation) included

**Comment by asmod3us** [comment]
`vars` and `reqvqrs` - is that vars=variables as in optional vs reqvars=required variables as in "we need this to set up based on the blueprint"?

**Comment by Ornias1993** [comment]
@asmod3us Yes thats the meaning of reqvars.

vars: variables to be loaded from config.yml
reqvars: of which the following are required before install proceeds

**Pull request merged**